runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java

```diff
@@ -128,7 +128,15 @@ public Object getPayload(CorfuRuntime runtime) {
                             throw throwable;
                         } finally {
                             serializedBuf.release();
-                            data = null;
+                            // LR doesn't read deserialized data. So in the few cases that the data is deserialized, LR
+                            // still reads the data part of LogData. Setting "nullifyDataOnGetPayload" param for LR will
+                            // ensure that no data is skipped from being seen by LR.
+                            // Required for the registry table by all the replication models and for the tables
```

Comment:
Can you remind me why this happens also for LogicalGroup ad RoutingQueue tables? I was under the impression this was only required for registryTable and perhaps status tables. Is routingQueues case because we unwrap the outer part? what about logical group?

Reply:
Yes for the routingQueues.
We don't need this fix for logical group. I got confused and was thinking we somewhere deserialize to check the logical group in the metadata and if an FSM needs to send it.
But I correct myself..the configManager takes care of finding the correct streams to replicate. After that there is no need to look at the metadata. I'll remove the mention of logical group model here.